Philips.CodeAnalysis.Common/SyntaxNodeSingleDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,87 @@
+﻿// © 2023 Koninklijke Philips N.V. See License.md in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Philips.CodeAnalysis.Common
+{
+	public abstract class SingleDiagnosticAnalyzer<T> : SingleDiagnosticAnalyzer where T : SyntaxNode
```

Comment:
Please rename this file to `SingleDiagnosticAnalyzer{T}.cs` to prevent the dogfood. Might need work on the dogfood reporting Analyzer.

Reply:
Will do on next round later today